### ISSUE TITLE
Xdrip not uploading readings any more "marking record:failed"

### ISSUE THREAD

**Issue by bionictom** [nightscout/cgm-remote-monitor#7431]
Hi, i have been having problems with xdrip not pushing my numbers to nightscout for a few days now.

Error message says:
"unable to do REST API Upload: Failed to connect to ..... marking record:failed".

AAPS is still pushing treatments, and i've cleard the devicestatus on nightscout.

**Comment by psonnera** [comment]
Can you post your xDrip+ uploader status (hiding personal information) here?

**Comment by bionictom** [comment]
Crazily enough, just as I was about to take screenshots, i noticed that the error message was gone and all of the readings are now on nightscout, without me changing anything in the last 3 days. It fixed itself.

**Comment by bionictom** [comment]
Actually, the problem re-appeared. The old data got unploaded, but it stopped again, 4 hours ago.

**Comment by bionictom** [comment]
![IMG_20220521_211258](https://user-images.githubusercontent.com/28786121/169666168-99d4d721-8e11-4312-839a-3a903848e76b.jpg)

**Comment by Geoff85** [comment]
I'm having this issue too. Has anyone got a solution for it?

**Comment by Root-Core** [comment]
Same issue, I installed the latest update and it's still not working. So restarting the app does nothing and it is present in the current release. AAPS is working via local broadcasts.

**Comment by psvtg** [comment]
Same problem now

**Comment by herzogmedia** [comment]
Hi, I had that message a few times as well and in most cases is had to do with the mongo service not working. If you're on atlas you could try to login there and check if everything is alright

**Comment by acember** [comment]
I'm having this problem now and just read this whole thread, but it doesn't look like anyone figured out how to fix it... :/

My system status on xDrip says "Unable to do REST API Upload: Unexpected baseURI 

But the REST API is typed correctly as is the URL, so I don't know what else to do to get NS running again.

**Comment by psonnera** [comment]
@acember can you open your Nightscout site page? Is it functional?

**Comment by acember** [comment]
@psonnera Yeah, the site itself is functional -- there just isn't any data there. But I can, for example, add a dummy treatment or something like that.

**Comment by acember** [comment]
Update: It turns out that if you're using ns.10be.de as your server, you need to add _/api/v1/_ to your API secret and URL to form the xDrip Base URL. So, for example, my URL is https://sputnik4abby.ns.10be.de/ , so I used:

https://MY_API_SECRET@https://sputnik4abby.ns.10be.de/api/v1/

Hopefully that helps someone else!
